commons/src/scanner/symbol/iterator/mod.rs

```diff
@@ -0,0 +1,504 @@
+//! Contains the [`SymbolIterator`], and all related functionality
+//! that is used to step through the [`Symbol`]s retrieved from the [`Scanner`](crate::scanner::Scanner).
+
+use std::borrow::BorrowMut;
+
+use super::{Symbol, SymbolKind};
+
+mod matcher;
+mod root;
+
+pub use itertools::*;
```

Comment:
I would re-export `itertools`, because I assume mostly our `unimarkup` crates will depend on `commons`.
This makes it easy to keep the dependency of `itertools` in `commons`,
and all our other crates using `commons` automatically get features available in `itertools`.
Having to manage `itertools` in every crate using `commons` seems more painful.

Reply:
There's also discussion about compatibility if we use different versions of `itertools` in for example `parser` and in `commons`. This solves that, but also makes it so that public api of `commons` is extended by that of `itertools`. I don't know what is best, so we can leave it as is for now.